packages/components/src/internal/components/ontology/OntologyTreeSearchContainer.tsx

```diff
@@ -1,25 +1,81 @@
 
 import { ConceptModel, OntologyModel, PathModel } from './models';
-import { fetchAlternatePaths } from './actions';
+import { fetchAlternatePaths, getOntologyDetails } from './actions';
 
 const CONCEPT_CATEGORY = 'concept';
 const SEARCH_LIMIT = 20;
 
+interface OntologySearchInputProps {
+    ontologyId: string;
+    initCode?: string; //initial concept code
+    inputName: string;
+    className: string;
+    searchPathClickHandler: (code: string) => void;
+    showBrowserHandler?: () => void;
+    fieldName?: string;
+}
+
+export const OntologySearchInput: FC<OntologySearchInputProps> = memo(props => {
+    const {ontologyId, searchPathClickHandler, ...rest} = props;
+    const [ontologyModel, setOntologyModel] = useState<OntologyModel>();
+    const [error, setError] = useState<string>();
+
+    useEffect(() => {
+        if (ontologyId) {
+            getOntologyDetails(ontologyId)
+                .then((ontology: OntologyModel) => {
+                    setOntologyModel(ontology);
+                })
+                .catch(() => {
+                    setError('Error: unable to load ontology concept information for ' + ontologyId + '.');
+                    // setSelectedOntologyId(undefined);
+                });
+        } else {
+            setOntologyModel(undefined);
+        }
+    }, [ontologyId, setOntologyModel, setError]);
+
+    const onSearchClickHandler = useCallback((path: PathModel) => {
+        searchPathClickHandler(path.code)
+    }, [searchPathClickHandler]);
+
+    return (
+        <>
+            <Alert>{error}</Alert>
+            {ontologyModel && <OntologyTreeSearchContainer
+                {...rest}
+                ontology={ontologyModel}
+                searchPathClickHandler={onSearchClickHandler}
+            />}
+        </>
+    );
+});
+
 interface OntologyTreeSearchContainerProps {
     ontology: OntologyModel;
+    inputName?: string;
+    initCode?: string;
+    className?: string;
     searchPathClickHandler: (path: PathModel, isAlternatePath?: boolean) => void;
+    showBrowserHandler?: () => void;
+    fieldName?: string;
 }
 
 export const OntologyTreeSearchContainer: FC<OntologyTreeSearchContainerProps> = memo(props => {
-    const { ontology, searchPathClickHandler } = props;
+    const { ontology, searchPathClickHandler, inputName = "concept-search", className = "form-control", initCode = "", showBrowserHandler, fieldName } = props;
     const [isFocused, setIsFocused] = useState<boolean>();
-    const [searchTerm, setSearchTerm] = useState<string>();
+    const [searchTerm, setSearchTerm] = useState<string>(initCode);
     const [searchHits, setSearchHits] = useState<ConceptModel[]>();
     const [totalHits, setTotalHits] = useState<number>();
     const [error, setError] = useState<string>();
+    const [showResults, setShowResults] = useState<boolean>(false);
+
+    useEffect(() => {
+        setSearchTerm(initCode);
+    }, [initCode, setSearchTerm]);
```

Comment:
from what I see above, it looks like `initCode` is already being set as the initial `searchTerm`. or is this the case that handles when the user types into the input field to change the `initCode` prop value?

Reply:
it looks like `onSearchChange` is what handles the user input change, so I'm unsure if this `useEffect` is needed.